config/confluent_kafka_topic/config.go

```diff
@@ -0,0 +1,48 @@
+package confluent_kafka_topic
+
+import (
+	"fmt"
+
+	xpref "github.com/crossplane/crossplane-runtime/pkg/reference"
+	xpresource "github.com/crossplane/crossplane-runtime/pkg/resource"
+	"github.com/crossplane/upjet/pkg/config"
+	"github.com/crossplane/upjet/pkg/resource"
+)
+
+// Constants for custom Extractor function
+var (
+	selfPackagePath     = "github.com/crossplane-contrib/provider-confluent/config/confluent_kafka_topic"
+	extractResourceIDFn = selfPackagePath + ".ExtractResourceID()"
+)
+
+// Configure configures individual resources by adding custom ResourceConfigurators.
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("confluent_kafka_topic", func(r *config.Resource) {
+		// We need to override the default group that upjet generated for
+		r.ShortGroup = "confluent"
+		r.UseAsync = true
+		r.Kind = "KafkaTopic"
+
+		// Allows us to reference managedResource ID via spec.forProvider.managedResource.id
+		r.References["kafka_cluster.id"] = config.Reference{
+			Type: "Cluster",
+		}
+
+		// Allows us to reference managedResource ID via spec.forProvider.principal.idSelector
+		r.References["principal"] = config.Reference{
```

Comment:
Was this a leftover copy and paste? I don't see `principal` listed as a field on the terraform docs for this resource, ref: https://registry.terraform.io/providers/confluentinc/confluent/1.55.0/docs/resources/confluent_kafka_topic

Reply:
> Was this a leftover copy and paste? I don't see `principal` listed as a field on the terraform docs for this resource, ref: https://registry.terraform.io/providers/confluentinc/confluent/1.55.0/docs/resources/confluent_kafka_topic

You're absolutely right. Somehow i mixed that up with the credentials parameters, but this is obviously something different. Sorry for that.